push-agents/apns-agent/src/main/java/org/aerogear/push/apns/ios/PushyApnsSender.java

```diff
@@ -0,0 +1,214 @@
+package org.aerogear.push.apns.ios;
+
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.turo.pushy.apns.ApnsClient;
+import com.turo.pushy.apns.ApnsClientBuilder;
+import com.turo.pushy.apns.PushNotificationResponse;
+import com.turo.pushy.apns.util.ApnsPayloadBuilder;
+import com.turo.pushy.apns.util.SimpleApnsPushNotification;
+import io.netty.util.concurrent.Future;
+import org.aerogear.push.apns.helper.InternalUnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.iOSVariant;
+import org.jboss.aerogear.unifiedpush.message.Message;
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.message.apns.APNs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.util.Collection;
+import java.util.Map;
+
+public class PushyApnsSender {
+
+    private final Logger logger = LoggerFactory.getLogger(PushyApnsSender.class);
+
+    public static final String KAFKA_INVALID_TOKEN_TOPIC = "agpush_invalidToken";
+
+
+    /**
+     * Topic to which a "success" message will be sent if a token was accepted and "failure" message otherwise.
+     */
+    public static final String KAFKA_APNS_TOKEN_DELIVERY_METRICS_TOPIC = "agpush_apnsTokenDeliveryMetrics";
+    public static final String KAFKA_APNS_TOKEN_DELIVERY_SUCCESS = "agpush_apnsTokenDeliverySuccess";
+    public static final String KAFKA_APNS_TOKEN_DELIVERY_FAILURE = "agpush_pnsTokenDeliveryFailure";
+
+    public static final String CUSTOM_AEROGEAR_APNS_PUSH_HOST = "custom.aerogear.apns.push.host";
+    public static final String CUSTOM_AEROGEAR_APNS_PUSH_PORT = "custom.aerogear.apns.push.port";
+
+
+
+    public void sendPushMessage(final Variant variant, final Collection<String> tokens, final UnifiedPushMessage pushMessage, final String pushMessageInformationId, final NotificationSenderCallback senderCallback) {
+
+        // no need to send empty list
+        if (tokens.isEmpty()) {
```

Comment:
The method is public, so it is exposed to uncontrolled calls. I would suggest checking for null value as well.